### PR TITLE
openstack: show details on create_volume failure

### PR DESCRIPTION
**Pull request by ktdreyer**
##### SUMMARY
Prior to this change, if we failed to create an OpenStack volume (for example, because a user's volume quota is full), we simply displayed the vague error message "Error in creating volume", which is the simple `str` representation of the `HttpException` from `create_volume()`. Users had difficulty understanding why the volume creation failed.

Cinder will provide us more detailed information about the creation failure, and openstacksdk captures this in the `HttpException`'s `.details` attribute.

If we encounter this exception for `create_volume()`, display this "details" string to the user, so it's easier to understand what went wrong.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
os_volume

### CODE REVIEW

File: lib/ansible/modules/cloud/openstack/os_volume.py [hunk 1]
```diff
@@ -111,9 +111,14 @@ def _present_volume(module, cloud):
...
-    volume = cloud.create_volume(
-        wait=module.params['wait'], timeout=module.params['timeout'],
-        **volume_args)
+    try:
+        volume = cloud.create_volume(
+            wait=module.params['wait'], timeout=module.params['timeout'],
+            **volume_args)
+    except sdk.exceptions.HttpException as e:
+        if hasattr(e, 'details'):
+            module.fail_json(msg=str(e.details))
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/openstack/os_volume.py]:
```suggestion
            module.fail_json(msg=to_native(e.details))
```

**Review by Akasurde — CHANGES_REQUESTED** [review]
Could you please missing `RETURNS` section in this module ? Thanks.

File: lib/ansible/modules/cloud/openstack/os_volume.py (hunk 1 above)
Comment (ktdreyer) [reply lib/ansible/modules/cloud/openstack/os_volume.py]: Thanks @Akasurde . I've amended my commit and force-pushed.

**Comment by ktdreyer** [comment]
@Akasurde , I'm ok updating `RETURNS` in a separate PR, since that is orthogonal to this change.

**Review by Akasurde — APPROVED** [review]

**Comment by Akasurde** [comment]
@ktdreyer Sure. no problem.

**Comment by ktdreyer** [comment]
I've added the `RETURNS` doc separately at https://github.com/ansible/ansible/pull/49559

**Comment by ktdreyer** [comment]
There is actually a better way to do this across the board for all openstack modules. I'll open a new PR.

**Comment by ktdreyer** [comment]
new PR at https://github.com/ansible/ansible/pull/49573

**Pull request closed without merging**